.github/workflows/main.yml

```diff
@@ -1,40 +1,57 @@
 name: master
 on:
   push:
-    branches:
-      - master
+    # branches:
+    #   - master
 permissions:
   contents: read
   id-token: write
 concurrency:
   cancel-in-progress: false
   group: "${{ github.workflow }}-${{ github.ref }}"
 jobs:
-  upload-aws-backup-elastio-integration:
-    runs-on:
-      - ubuntu-latest
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      ci: "${{ steps.filter.outputs.ci }}"
+      aws-backup-elastio-integration: "${{ steps.filter.outputs.aws-backup-elastio-integration }}"
+      elastio-s3-changelog: "${{ steps.filter.outputs.elastio-s3-changelog }}"
     steps:
       - name: Checkout repository
-        uses: "actions/checkout@v3"
+        uses: actions/checkout@v4
         with:
-          fetch-depth: '30'
-          token: "${{ secrets.SECRET_TOKEN }}"
+          token: ${{ secrets.SECRET_TOKEN }}
       - name: Check if files were modified
-        id: changes
-        uses: "dorny/paths-filter@v2"
+        id: filter
+        uses: "dorny/paths-filter@v3"
         with:
           filters: |
-            filechange:
+            ci:
+              - '.github/workflows/**'
+            aws-backup-elastio-integration:
               - 'aws-backup-elastio-integration/**'
+            elastio-s3-changelog:
+              - 'elastio-s3-changelog/**'
+  upload-aws-backup-elastio-integration:
+    runs-on:
+      - ubuntu-latest
+    needs: changes
+    if: >-
+      needs.changes.outputs.aws-backup-elastio-integration == 'true'
+        && github.ref == 'refs/heads/master'
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: '30'
+          token: ${{ secrets.SECRET_TOKEN }}
```

Comment:
I don't think we need any secret token here and any overrides to `actions/checkout` because we don't use any submodules in the repo and we don't use any private Elastio code here as well

Reply:
got it. Removing